pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -114,6 +122,24 @@ func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.Syn
 		}
 	}
 
+	// In case a checkProfileRejection func was used, status will be updated accordingly
+	if c.checkProfileRejectionFn != nil {
+		isRejected, rejectMsg, err := c.checkProfileRejectionFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
+		if err != nil {
+			return err
+		}
+		// if profile transition is rejected, set status with rejection message
+		if isRejected {
+			_, err = c.updateStatus(
+				ctx,
+				false, true, // not progressing, complete=True
```

Comment:
Yes, I understand the reasoning behind the `""` as it is accepted value. But this is a rejected value and I think we should notify the user somehow that the configuration was not resolved and will not be resolved. Either via `complete=False` or degrading the operator? 

Let's start a discussion on this topic to get more opinion on this.

Reply:
As discussed in [1], we could go ahead and degrade the controller in case of encountering rejection(s).
+ add a suitable message in the error related to recovering i.e. to rectify the profile (as suggested by @p0lyn0mial).

[1] https://coreos.slack.com/archives/C0392N52Q5P/p1653931012883689?thread_ts=1653567121.768759&cid=C0392N52Q5P

Reply:
Sample output with current implementation (degrading upon rejections).

```
    - lastTransitionTime: "2022-06-03T07:02:13Z"
      reason: ProfileUpdateProhibited
      status: Unknown
      type: WorkerLatencyProfileProgressing
    - lastTransitionTime: "2022-06-03T07:02:13Z"
      reason: ProfileUpdateProhibited
      status: Unknown
      type: WorkerLatencyProfileComplete
    - lastTransitionTime: "2022-06-03T07:02:13Z"
      message: rejected update from "LowUpdateSlowReaction" to "Default" latency profile
        as extreme profile transition is unsupported, please set nodes.config.openshift.io/cluster
        object to worker latency profile "LowUpdateSlowReaction" (or any other profile
        apart from rejected "Default") in order to recover from this error
      reason: SyncError
      status: "True"
      type: WorkerLatencyProfileDegraded
```